src/Orleans.Core/Runtime/CallbackData.cs

```diff
@@ -54,7 +57,7 @@ public void OnTargetSiloFail()
         {
             if (this.IsCompleted)
                 return;
-
+            EventSourceUtils.EmitEvent(this.Message, OrleansCallBackDataEvent.Log.OnTargetSiloFail);
```

Comment:
This pattern is very expensive. Each time a call to `EmitEvent` is made, it will allocate a new `Action` for this second parameter.

If we cache the `Action` in a field on the EventSource class then the allocation goes away and it should be much cheaper.

Reply:
I think if we just fix this then it can be merged